Technical Analytics: Milestone 1 - Add Feature Flag Statuses and Ability To Sync Them to Cache Store

kkmurerwa opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
When merged, this PR will:
- Create a new `FeatureFlagConstants.kt` file to contain all feature flags.
- Create feature flag sync status trackers for each existing feature flag.
- Modify the `PlatformParameterSyncUpWorker` to allow status trackers of all feature flags to be synced and cached in the PersistentCacheStore.
- Merge the `TestBooleanPlatformParameter`, `TestStringPlatformParameter` and the `TestIntegerPlatformParameter` files into the `TestPlatformParameterConstants` file.
- Add syncing with the web for the `EnableDownloadsSupport`, `EnableEditAccountsOptionsUi`, `EnableSpotlightUi`, `EnableExtraTopicTabsUi`, `EnableInteractionConfigChangeStateRetention`, and `EnableAppAndOsDeprecation` feature flags.
- Write tests for the changes made.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

kkmurerwa committed:
feat: Add a FeatureFlagConstants file to host feature flags

kkmurerwa committed:
feat: Add constant names for easy retrieval of the feature flags from persistent cache store

kkmurerwa committed:
feat: Add newly status flags to the Platform Parameter dependency modules

kkmurerwa committed:
feat: Add ability to insert sync statuses to the cache store

kkmurerwa committed:
feat: Add ability to save flag statuses for every feature flag

kkmurerwa committed:
Merge branch 'oppia:develop' into technical-analytics-milestone-1

kkmurerwa committed:
fix: Fix linting and styling checks

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

adhiamboperes commented on utility/src/main/java/org/oppia/android/util/platformparameter/FeatureFlagConstants.kt:
```diff
@@ -0,0 +1,360 @@
+ *  - Feature Flag Name - The name begins with Enable_
+ *  - Feature Flag Default Value
+ *  - Feature Flag Status - A boolean that keeps track of whether the feature flag
+ *  has been synced with Oppia Web or not.
```
Nit
```suggestion
 *  has been synced with Oppia Web.
```

kkmurerwa replied:
This has been modified.

adhiamboperes commented on utility/src/main/java/org/oppia/android/util/platformparameter/FeatureFlagConstants.kt:
```diff
@@ -0,0 +1,360 @@
+annotation class EnablePerformanceMetricsCollection
+
+/** Name of the platform parameter that controls whether to record performance metrics. */
+const val ENABLE_PERFORMANCE_METRICS_COLLECTION = "enable_performance_metrics_collection"
```
Change `platform parameter` to `feature flag`

kkmurerwa replied:
I have changed this and other instances in the FeatureFlagConstants file.

adhiamboperes commented on utility/src/main/java/org/oppia/android/util/platformparameter/FeatureFlagConstants.kt:
```diff
@@ -0,0 +1,360 @@
+ * Qualifier for the platform parameter that controls whether to animate the continue button
+ * interaction and navigation items. This is used to disable the animation during testing because
+ * Espresso has known problems while testing views that contain animations.
+ */
```
Change `platform parameter` to `feature flag`

kkmurerwa replied:
See above comment.

adhiamboperes commented on testing/src/main/java/org/oppia/android/testing/platformparameter/TestStringPlatformParameter.kt:
```diff
@@ -22,3 +22,18 @@ const val TEST_STRING_PARAM_DEFAULT_VALUE = "test_string_param_default_value"
+
+/**
+ * Name for the test string platform parameter. Only used in tests related to platform parameter.
+ */
```
This KDoc is identical to the one above, and does not provide clarity regarding the differnces between the two constants.

kkmurerwa replied:
I have updated this comment and other comments in this file as well that are not very clear.

adhiamboperes commented on testing/src/main/java/org/oppia/android/testing/platformparameter/TestStringPlatformParameter.kt:
```diff
@@ -22,3 +22,18 @@ const val TEST_STRING_PARAM_DEFAULT_VALUE = "test_string_param_default_value"
+/**
+ * Server value for the test string platform parameter. Only used in tests related to platform parameter.
+ */
+const val TEST_STRING_PARAM_SYNC_STATUS_FLAG_SERVER_VALUE = true
```
It may be useful to ensure these values follow the same naming pattern as their prod counterparts. Please also ensure that the KDocs are clear.

kkmurerwa replied:
I have updated the naming patterns of these constants.

adhiamboperes commented on domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:
```diff
@@ -132,6 +135,27 @@ class PlatformParameterSyncUpWorker private constructor(
+
+      val syncStatusParamKey = "flag_" + param.name + "_is_server_provided"
+
+      // Add it to the modified list
```
It is beneficial instead, to explain why the `paramSyncStatusTracker.boolean` is being updated. At a first glance, I am not able to understand what that part of the code does.

kkmurerwa replied:
I added a comment, but I am not sure if it's clear enough now. PTAL.

adhiamboperes commented on domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:
```diff
@@ -111,11 +111,14 @@ class PlatformParameterSyncUpWorker private constructor(
         val platformParameterList = parseNetworkResponse(responseBody)
-        if (platformParameterList.isEmpty()) {
+
+        // Add boolean flags
```
This is not necessary

kkmurerwa replied:
I'm assuming you meant the comment. I have removed it.

adhiamboperes commented on domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:
```diff
@@ -132,6 +135,27 @@ class PlatformParameterSyncUpWorker private constructor(
     }
   }
 
+  private fun addSyncStatusFlags(
```
I think this could read better as an extension function, so that you can have:

`platformParameterList.addSyncStatusFlags()`

kkmurerwa replied:
I created an extension function in the same Kotlin file. I believe this was the best way to do it. Let me know if I could have used another approach.

adhiamboperes reviewed (changes requested):
Thanks @kkmurerwa!

I have left comments inline, mostly calling for clarity.

kkmurerwa committed:
fix: Fix comments made on the previous review

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa commented:
Hi @adhiamboperes. I have made the suggested changes. PTAL.

adhiamboperes commented on domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:
```diff
@@ -132,6 +134,25 @@ class PlatformParameterSyncUpWorker private constructor(
     }
   }
 
+  private fun List<PlatformParameter>.addSyncStatusFlags(): List<PlatformParameter> {
+    val modifiedList = mutableListOf<PlatformParameter>()
+
+    for (param in this) {
+      modifiedList.add(param)
+      val syncStatusParamKey = "flag_" + param.name + "_is_server_provided"
+
+      val paramSyncStatusTracker = PlatformParameter.newBuilder().setName(syncStatusParamKey)
+      // Update the boolean status of the derived syncStatusParamKey to true. This is necessary
+      // since sync status flags are local only and don't have a preset value.
+      paramSyncStatusTracker.boolean = true
+      paramSyncStatusTracker.build()
+
+      modifiedList.add(paramSyncStatusTracker.build())
+    }
+
+    return modifiedList
+  }
```
On line 150, you are building `paramSyncStatusTracker` a second time after building it in line 148.
Does my suggested comment update convey the correct explanation?

```suggestion
 private fun List<PlatformParameter>.addSyncStatusFlags(): List<PlatformParameter> {
    val modifiedList = mutableListOf<PlatformParameter>()

    for (param in this) {
      modifiedList.add(param)
      val syncStatusParamKey = "flag_" + param.name + "_is_server_provided"

      val paramSyncStatusTracker = PlatformParameter.newBuilder().apply { 
        name =syncStatusParamKey
        boolean = true // Indicates that sync status flags are up to date since they are local only
      }
        .build()
      modifiedList.add(paramSyncStatusTracker)
    }

    return modifiedList
  }
```

kkmurerwa replied:
Ah, I see. I have merged (and resolved) your suggestion since it is more concise and easier to understand.

adhiamboperes reviewed (changes requested):
Thanks @kkmurerwa. Just one other concern.

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
fix: Fix nit

Co-authored-by: Adhiambo Peres <59600948+adhiamboperes@users.noreply.github.com>

kkmurerwa commented:
@adhiamboperes I have addressed the comments made. PTAL

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
fix: Fix lint issues causing build failure

kkmurerwa committed:
fix: Move test sync status flag booleans to the TestBooleanPlatformParameter file

adhiamboperes commented on testing/src/main/java/org/oppia/android/testing/platformparameter/TestBooleanPlatformParameter.kt:
```diff
@@ -22,3 +22,22 @@ const val TEST_BOOLEAN_PARAM_DEFAULT_VALUE = false
  * Server value for the test boolean platform parameter. Only used in tests related to platform parameter.
  */
 const val TEST_BOOLEAN_PARAM_SERVER_VALUE = true
+
+/**
+ * Feature flag sync status name for the test string platform parameter. This name helps retrieve
+ * the sync status of the [TestStringParam]. Only used in tests related to platform parameter.
+ */
+const val FLAG_TEST_STRING_PARAM_IS_SERVER_PROVIDED =
+  "flag_test_string_param_name_is_server_provided"
+
+/**
+ * Default value for the feature flag sync status tracker for the [TestStringParam]. Only used in
+ * tests related to platform parameter.
+ */
+const val FLAG_TEST_STRING_PARAM_IS_SERVER_PROVIDED_DEFAULT_VALUE = false
+
+/**
+ * Server value for the test string platform parameter. Only used in tests related to platform
+ * parameter.
+ */
+const val TEST_STRING_PARAM_SYNC_STATUS_FLAG_SERVER_VALUE = true
```
As discussed offline, these appear to be test values corresponding to the sync status flags, which are in production, boolean values. The naming implies they are strings, so th's a bit confusing.
Also, I don't see any current usagaes of these values, unless I'm missing something.

kkmurerwa replied:
Based on the discussion had with Ben, I have merged the constants in these files to one constants file. PTAL.

adhiamboperes reviewed (commented):
Thanks @kkmurerwa. I am assigning this over to @BenHenning for codeowners since new patterns are being introduced.

@BenHenning, PTAL.

kkmurerwa committed:
chore: Move all test constants from individual type-organized files to a common constants file

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
fix: Fix failing text file checks

kkmurerwa committed:
Merge branch 'technical-analytics-milestone-1' of github.com:kkmurerwa/oppia-android into technical-analytics-milestone-1

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

adhiamboperes reviewed (approved):
This LGTM, thanks @kkmurerwa

BenHenning commented on domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:
```diff
@@ -132,6 +134,24 @@ class PlatformParameterSyncUpWorker private constructor(
   }
 
+  private fun List<PlatformParameter>.addSyncStatusFlags(): List<PlatformParameter> {
+    val modifiedList = mutableListOf<PlatformParameter>()
+
+    for (param in this) {
+      modifiedList.add(param)
+      val syncStatusParamName = "flag_" + param.name + "_is_server_provided"
+
+      val paramSyncStatusTracker = PlatformParameter.newBuilder().apply {
+        name = syncStatusParamName
+        boolean = true // Indicates that sync status flags are up to date since they are local only
+      }
+        .build()
+      modifiedList.add(paramSyncStatusTracker)
+    }
+
+    return modifiedList
```
```suggestion
    return this + map { param ->
      PlatformParameter.newBuilder().apply {
        name = "flag_" + param.name + "_is_server_provided"
        boolean = true // Indicates that sync status flags are up to date since they are local only
      }.build()
    }
```

Can be made more idiomatic & simplified.

kkmurerwa replied:
This is more concise but is no longer a part of the code with the new changes that I have made.

BenHenning commented on domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:
```diff
@@ -132,6 +134,24 @@ class PlatformParameterSyncUpWorker private constructor(
+
+      val paramSyncStatusTracker = PlatformParameter.newBuilder().apply {
+        name = syncStatusParamName
+        boolean = true // Indicates that sync status flags are up to date since they are local only
```
Rather than creating a new PlatformParameter, have you considered adding a property to the PlatformParameter structure to indicate whether it's been synced from the server?

The two main potential problems I see with this approach are:
- It's possible for other parameters to theoretically conflate on the name (since we're generating a name, another parameter could try to use that exact same name). Strings aren't reliable for representing types for this reason.
- It doubles the number of parameters we need to maintain for all parameters that can be remotely synced.

kkmurerwa replied:
I worked on an implementation that uses this approach and updated most of the code. PTAL and let me know if this is what you had in mind.

BenHenning reviewed (commented):
Thanks @kkmurerwa. I did have a few thoughts--PTAL.

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
feat: Add a isSynced variable in the platform parameter variable to simplify sync monitoring

kkmurerwa committed:
fix: Fix failing EventBundleCreator tests

kkmurerwa commented:
Hi @BenHenning. I have addressed the suggested changes. PTAL

BenHenning commented on utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterValue.kt:
```diff
@@ -9,16 +9,23 @@ import org.oppia.android.app.model.PlatformParameter
           get() = defaultValue
+
+        override val isSynced: Boolean
+          get() = defaultIsSynced
```
```suggestion
        override val value = defaultValue
        override val isSynced = defaultIsSynced
```

Getters aren't really needed here, so this can be simplified a bit.

kkmurerwa replied:
I have updated this and all other instances to remove getters.

BenHenning commented on utility/src/test/java/org/oppia/android/util/logging/KenyaAlphaEventBundleCreatorTest.kt:
```diff
@@ -1583,6 +1583,7 @@ class KenyaAlphaEventBundleCreatorTest {
       val enableFeature = enableLoggingLearnerStudyIds
       return object : PlatformParameterValue<Boolean> {
```
Here & everywhere else in this PR: could we use createDefaultParameter() instead of creating a custom implementation of the class? It'd probably simplify future changes to the interface, too, by avoiding needing to change a bunch of places.

kkmurerwa replied:
createDefaultParameter is the default approach for all test object creations now.

BenHenning commented on domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonImpl.kt:
```diff
@@ -26,6 +26,8 @@ class PlatformParameterSingletonImpl @Inject constructor() : PlatformParameterSi
       override val value: String
         get() = parameter.string
+      override val isSynced: Boolean
+        get() = parameter.isSynced
```
```suggestion
      override val value = parameter.string
      override val isSynced = parameter.isSynced
```

Can be simplified. Ditto for the other updates in this class.

kkmurerwa replied:
This has been updated as part of the "removing getters" changes.

BenHenning commented on domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt:
```diff
@@ -1358,6 +1358,7 @@ class ProfileManagementControllerTest {
       val enableFeature = enableLearnerStudyAnalytics
       return object : PlatformParameterValue<Boolean> {
         override val value: Boolean = enableFeature
+        override val isSynced: Boolean = true
```
Do this & the next parameter need to be set to isSynced to true? Why? It seems like all other tests are using a default not-synced value.

kkmurerwa replied:
Since we no longer use booleans, I have switched to the SYNC_STATUS_UNSPECIFIED as the defualt status for these.

BenHenning commented on model/src/main/proto/platform_parameter.proto:
```diff
@@ -18,6 +18,8 @@ message PlatformParameter {
     // Indicates a string-typed platform parameter.
     string string = 4;
   }
+  // Indicates that the platform parameter is synced
+  bool is_synced = 5;
```
For future proofing this, we should probably use an enum instead of a bool. E.g.:

```proto
enum SyncStatus {
  // Indicates that the sync status isn't yet known.
  SYNC_STATUS_UNSPECIFIED = 0;

  // Indicates that the parameter isn't yet synced with the remote server.
  NOT_SYNCED_FROM_SERVER = 1;

  // Indicates the parameter's value has been synced with the remote server.
  SYNCED_FROM_SERVER = 2;
}
```

Then it could be used as so:

```suggestion

  // Indicates the parameter's remote sync status.
  SyncStatus sync_status = 5;
```

This provides a few advantages:
- An enum is more descriptive.
- We get a proper 'default' value (unspecified vs. assuming 'false').
- We can add other sync types in the future without needing to migrate the true/false state.

kkmurerwa replied:
I have switched to this approach. I still can't picture a scenario where the NOT_SYNCED_FROM_SERVER is applicable so it is currently unused but the other two are being used and are working perfectly.

BenHenning replied:
It's actually used by default. If you create a new PlatformParameter instance and don't set the sync status then it'll be 0 (SYNC_STATUS_UNSPECIFIED) by default, per proto3. That's actually why using 0 as the unspecified makes sense since it will automatically be used in cases when we do not know what value to use (hence, 'unspecified').

kkmurerwa replied:
I get it now. Thanks.

BenHenning commented on domain/src/test/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkerTest.kt:
```diff
@@ -322,6 +326,89 @@ class PlatformParameterSyncUpWorkerTest {
+    val platformParameterMap = platformParameterSingleton.getPlatformParameterMap()
+
+    // Previous integer sync status is still the same in the database
+    assertThat(platformParameterMap[TEST_INTEGER_PARAM_NAME]!!.isSynced).isEqualTo(false)
```
```suggestion
    assertThat(platformParameterMap[TEST_INTEGER_PARAM_NAME]?.isSynced).isFalse()
```

kkmurerwa replied:
We are no longer using booleans so this suggestion is no longer applicable.

BenHenning commented on domain/src/test/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkerTest.kt:
```diff
@@ -322,6 +326,89 @@ class PlatformParameterSyncUpWorkerTest {
+      .containsEntry(TEST_STRING_PARAM_NAME, expectedTestStringParameter)
+
+    // Is synced boolean of the platform parameter is true
+    assertThat(platformParameterMap[TEST_STRING_PARAM_NAME]!!.isSynced).isEqualTo(true)
```
```suggestion
    assertThat(platformParameterMap[TEST_STRING_PARAM_NAME]?.isSynced).isTrue()
```

kkmurerwa replied:
Same for this. Booleans are no longer being used so this change cannot be made.

BenHenning commented on domain/src/test/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkerTest.kt:
```diff
@@ -322,6 +326,89 @@ class PlatformParameterSyncUpWorkerTest {
+    testCoroutineDispatchers.runCurrent()
+
+    val workInfo = workManager.getWorkInfoById(request.id)
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
```
This can probably be removed since verifying this isn't actually relevant for this test (we verify pass/success of the worker in other tests--this test just cares about the outcome).

kkmurerwa replied:
I have removed it in the two tests where it appears.

BenHenning commented on domain/src/test/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkerTest.kt:
```diff
@@ -322,6 +326,89 @@ class PlatformParameterSyncUpWorkerTest {
+    testCoroutineDispatchers.runCurrent()
+
+    val workInfo = workManager.getWorkInfoById(request.id)
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.FAILED)
+
+    val exceptionMessage = fakeExceptionLogger.getMostRecentException().message
+    assertThat(exceptionMessage)
+      .isEqualTo(PlatformParameterSyncUpWorker.EMPTY_RESPONSE_EXCEPTION_MSG)
```
This can probably be removed since verifying this isn't actually relevant for this test (we verify pass/success of the worker in other tests--this test just cares about the outcome).

kkmurerwa replied:
I have removed this from the test.

BenHenning reviewed (changes requested):
Thanks @kkmurerwa! This is looking good. Just had a few follow-up comments (mostly small suggestions)--PTAL.

kkmurerwa committed:
feat: Make Sync status an enum

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
feat: Add feature flag names to all flags lacking one

kkmurerwa committed:
Merge branch 'technical-analytics-milestone-1' of github.com:kkmurerwa/oppia-android into technical-analytics-milestone-1

kkmurerwa committed:
fix: Fix failing check and linting issue

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa commented:
Hi @BenHenning. I have made the suggested changes and introduced others discussed during the team meeting. I have also updated the PR description to include all new changes made. PTAL

kkmurerwa committed:
merge: Merge from develop and fix conflicts

kkmurerwa committed:
chore: Remove unused feature flag

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
fix: Fix minor linting issues

BenHenning commented on utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterValue.kt:
```diff
@@ -18,14 +19,11 @@ interface PlatformParameterValue<T> {
     fun <T> createDefaultParameter(
       defaultValue: T,
-      defaultIsSynced: Boolean = false
+      defaultIsSynced: SyncStatus = SyncStatus.NOT_SYNCED_FROM_SERVER
```
```suggestion
      defaultSyncStatus: SyncStatus = SyncStatus.NOT_SYNCED_FROM_SERVER
```

kkmurerwa replied:
This has been updated.

BenHenning reviewed (approved):
Thanks @kkmurerwa & really sorry about the delayed review. The PR LGTM.

I have 2 follow-up comments: one following up to a point you made (no action needed), and the other to address a small nit in the code that I noticed. Once both are addressed, feel free to self-resolve those two and merge the PR once CI is passing.

BenHenning commented:
I'll also update to the latest develop so that CI has time to run before you're able to take another look at this @kkmurerwa.

BenHenning committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
Merge branch 'develop' into technical-analytics-milestone-1

kkmurerwa committed:
chore: Modify the PlatformParameterValue default sync status name
